aQute.libg/test/aQute/lib/io/IOTest.java

```diff
@@ -54,4 +55,70 @@ public void testDestDirIsChildOfSource() throws Exception {
 		} catch (IllegalArgumentException e) {
 		}
 	}
+
+	public void testIfCreateSymlinkOrCopyFileDependingOnOS() throws Exception {
+		File link = new File("generated/test/target.dat");
+
+		IO.delete(link);
+
+		assertFalse(link.exists() || Files.isSymbolicLink(link.toPath()));
+
+		link.getParentFile().mkdirs();
+
+		File source = new File("testresources/zipped.dat");
+
+		assertTrue(source.exists());
+
+		assertTrue(IO.createSymbolicLinkOrCopy(link, source));
+
+		if (IO.isWindows()) {
+			assertFalse(Files.isSymbolicLink(link.toPath()));
+		} else {
+			assertTrue(Files.isSymbolicLink(link.toPath()));
+		}
+	}
+
+	public void testOnlyCopyIfReallyNeededOnWindows() throws Exception {
+		if (IO.isWindows()) {
+			File link = new File("generated/test/target.dat");
+
+			IO.delete(link);
+
+			assertFalse(link.exists() || Files.isSymbolicLink(link.toPath()));
+
+			link.getParentFile().mkdirs();
+
+			File source = new File("testresources/zipped.dat");
+			assertTrue(source.exists());
+
+			assertTrue(IO.createSymbolicLinkOrCopy(link, source));
+
+			assertEquals(link.lastModified(), source.lastModified());
+			assertEquals(link.length(), source.length());
+
+			assertTrue(IO.createSymbolicLinkOrCopy(link, source));
+
+			assertEquals(link.lastModified(), source.lastModified());
+			assertEquals(link.length(), source.length());
+		}
+	}
+
+	public void testCreateSymlinkOrCopyWillDeleteOriginalLink() throws Exception {
+		File originalSource = new File("testresources/unzipped.dat");
+		File link = new File("generated/test/originalLink");
+
+		link.delete();
+
+		assertFalse(Files.isSymbolicLink(link.toPath()));
+
+		Files.createSymbolicLink(link.toPath(), originalSource.toPath());
+
+		File newSource = new File("testresources/zipped.dat");
+
+		IO.createSymbolicLinkOrCopy(link, newSource);
```

Comment:
I am not sure that is the right fix. I think the test is still somewhat interesting, but I think the expected results are in different. On either Windows or not windows, the link should be replaced.